crates/turbopack-ecmascript-plugins/src/transform/relay.rs

```diff
@@ -0,0 +1,72 @@
+use std::path::PathBuf;
+
+use async_trait::async_trait;
+use swc_core::{
+    common::{util::take::Take, FileName},
+    ecma::{
+        ast::{Module, Program},
+        visit::FoldWith,
+    },
+};
+use turbo_tasks_fs::FileSystem;
+use turbopack_ecmascript::{CustomTransformer, TransformContext};
+
+#[derive(Debug)]
+pub struct RelayTransformer {
+    config: swc_relay::Config,
+}
+
+impl RelayTransformer {
+    pub fn new(config: swc_relay::Config) -> Self {
+        Self { config }
+    }
+}
+
+#[async_trait]
+impl CustomTransformer for RelayTransformer {
+    async fn transform(
+        &self,
+        program: &mut Program,
+        ctx: &TransformContext<'_>,
+    ) -> Option<Program> {
+        let root = if let Ok(file_path) = ctx.file_path.await {
+            file_path
+                .fs
+                .root()
+                .await
+                .map_or(PathBuf::new(), |v| PathBuf::from(v.path.to_string()))
+        } else {
+            PathBuf::new()
+        };
```

Comment:
`transform` should return a `Result` and errors should be propagated instead of ignored.